### PR TITLE
Backport deployment using build matrix

### PR DESCRIPTION
**Pull request by ziyiwu9494**
Solve #2053

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,6 +20,7 @@ on:
   pull_request:
     branches:
       - master
```
Comment (ryanjulian) [inline .github/workflows/ci.yml:22]: we still get PRs to release branches, so the normal workflow still needs to run

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+  GARAGE_GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+  TRAVIS_COMMIT_RANGE: origin/${{ github.base_ref }}...origin/${{ github.head_ref }}
+  DOCKER_TAG_BASE: garage-ci-${{ github.run_id }}
+  OWNER: "rlworkgroup"
+  DOCKER_CACHE_REPO: "garage-ci"
+  MJKEY: ${{ secrets.MJKEY }}
+
+on:
+#  schedule:
```
Comment (ryanjulian) [inline .github/workflows/nightly.yml:13]: remove?

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+#    - cron: '0 9 * * *'
+  pull_request:
```
Comment (ryanjulian) [inline .github/workflows/nightly.yml:15]: if this is a nightly, why run on PRs?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ryanjulian) [reply .github/workflows/ci.yml:22]: if the two are incompatible, perhaps you also need a separate workflow for release-2020.06 PRs?

File: .github/workflows/nightly.yml (hunk 3 above)
Comment (ziyiwu9494) [reply .github/workflows/nightly.yml:15]: I'm just using this trigger to test

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ziyiwu9494) [reply .github/workflows/ci.yml:22]: This should still run for PRs and deployments

**Pull request closed without merging**